### DISCUSSION TITLE
文章正常编辑发布，但无法编辑草稿

### DISCUSSION THREAD

**Discussion by AuntYang** [Qexo/Qexo#333] [Q&A] [answered]
文章可以正常编辑、发布，但编辑草稿就报500，我按照wiki的5xx应对方法处理后仍然出现此问题。
![image](https://user-images.githubusercontent.com/49216233/228224728-dce7702c-f913-471b-b812-2c7630963eee.png)
我的源码库里确实存在Qexo创建的草稿，hpp监测token没问题
![image](https://user-images.githubusercontent.com/49216233/228224942-49bdca76-3a05-4b5e-a242-92ad466d244c.png)
想问问我该如何排除问题？从何开始检查？

**Comment by am-abudu** [comment, marked as answer]
需要排除问题。你现在有两个做法可以帮助我：
1. 提供Vercel函数(functions)日志
2. 提供你的博客仓库

**Reply by AuntYang** [reply]
1.Vercel函数(functions)日志
```
[WARNING]	2023-03-28T12:09:12.087Z	3865b32b-0fb8-4f13-bbbe-cfaca0e6638b	获取文件source/_drafts/web自动化测试（python selenium）.md
[ERROR]	2023-03-28T12:09:12.399Z	3865b32b-0fb8-4f13-bbbe-cfaca0e6638b	服务端错误: UnknownObjectException(404, {'message': 'Not Found', 'documentation_url': 'https://docs.github.com/rest/reference/repos#get-repository-content'}, {'server': 'GitHub.com', 'date': 'Tue, 28 Mar 2023 12:09:12 GMT', 'content-type': 'application/json; charset=utf-8', 'transfer-encoding': 'chunked', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': '', 'x-github-media-type': 'github.v3; format=json', 'x-github-api-version-selected': '2022-11-28', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '4989', 'x-ratelimit-reset': '1680008876', 'x-ratelimit-used': '11', 'x-ratelimit-resource': 'core', 'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'vary': 'Accept-Encoding, Accept, X-Requested-With', 'content-encoding': 'gzip', 'x-github-request-id': 'CF56:6E7A:23FB32:256FE1:6422D8E8'})
```
2.博客源码仓库
https://github.com/yepoda/hexo_source_code

**Reply by am-abudu** [reply]
初步看来是空格的问题，尽量不要用空格，我之后回尽可能解决这个问题

**Reply by AuntYang** [reply]
感谢您的回复，我测试后给您回复

**Reply by AuntYang** [reply]
Qexo编辑草稿时：
    文档名为：web自动化测试（python selenium），成功编辑草稿
    文档名为：web自动化测试（python+selenium），出现以上错误
Qexo能够获取源码仓库草稿（_drafts）中名为web自动化测试（python+selenium）的草稿
Qexo能够获取源码仓库文章（_posts）中名为web自动化测试（python+selenium）的文章

**Reply by am-abudu** [reply]
我已经复现了，我将尝试修复

**Reply by AuntYang** [reply]
值得注意的是，我将以上操作在myQexoDomain.com/posts.html页面再次尝试时均能够成功，而该页面显示的文章多于主页显示的文章，主页文章列表缺失了部分草稿文章

**Reply by am-abudu** [reply]
已修复，请测试。 8a8d11cac3cef4d866bdeaf4ac68a1ad09e27063

**Reply by AuntYang** [reply]
大佬，刚更新到dev分支，这是怎么了？
![image](https://user-images.githubusercontent.com/49216233/228246258-83d3d561-f004-48ed-babf-2146ad1510d7.png)

**Reply by am-abudu** [reply]
哇 我的锅 这里逻辑错了

**Reply by am-abudu** [reply]
改了一下，现在应该没问题了

**Reply by AuntYang** [reply]
ok了，草稿能正常编辑了，感谢！！

**Reply by AuntYang** [reply]
佬，删除草稿又来了😂点删除删不掉，文件名：web自动化测试（python selenium）
![image](https://user-images.githubusercontent.com/49216233/228250187-79e706aa-0982-462c-be24-e838ad3b1dac.png)

**Reply by am-abudu** [reply]
我试着解决了这个问题，你可以再试试

**Reply by AuntYang** [reply]
仍然无法删除草稿
[ERROR]	2023-03-28T13:56:54.840Z	77d147e3-8274-450b-af3c-66710640f14e	UnknownObjectException(404, {'message': 'Not Found', 'documentation_url': 'https://docs.github.com/rest/reference/repos#get-repository-content'}, {'server': 'GitHub.com', 'date': 'Tue, 28 Mar 2023 13:56:54 GMT', 'content-type': 'application/json; charset=utf-8', 'transfer-encoding': 'chunked', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': '', 'x-github-media-type': 'github.v3; format=json', 'x-github-api-version-selected': '2022-11-28', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '4912', 'x-ratelimit-reset': '1680012933', 'x-ratelimit-used': '88', 'x-ratelimit-resource': 'core', 'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'vary': 'Accept-Encoding, Accept, X-Requested-With', 'content-encoding': 'gzip', 'x-github-request-id': '8234:2F92:D8D79A:E1410C:6422F226'})

**Reply by am-abudu** [reply]
请提供文件名

**Reply by AuntYang** [reply]
web自动化测试（python selenium）

**Reply by am-abudu** [reply]
其实是删除了，但是列表没有更新吧。有个变量忘记改了

**Reply by AuntYang** [reply]
它还在hexo_source_code/source/_drafts/里.....

**Reply by am-abudu** [reply]
请确认是最新Dev分支，并清除缓存后重试。我本地测试已经通过

**Reply by AuntYang** [reply]
dev更新日志：
[WARNING]	2023-03-29T12:39:50.437Z	44e60557-b899-4bf2-96e7-8375069b01ea	获取更新成功: 2.6.3
报错日志：
[ERROR]	2023-03-29T12:39:54.231Z	db25e1ac-338f-472d-944c-bee0561afe14	UnknownObjectException(404, {'message': 'Not Found', 'documentation_url': 'https://docs.github.com/rest/reference/repos#get-repository-content'}, {'server': 'GitHub.com', 'date': 'Wed, 29 Mar 2023 12:39:54 GMT', 'content-type': 'application/json; charset=utf-8', 'transfer-encoding': 'chunked', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': '', 'x-github-media-type': 'github.v3; format=json', 'x-github-api-version-selected': '2022-11-28', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '4972', 'x-ratelimit-reset': '1680096378', 'x-ratelimit-used': '28', 'x-ratelimit-resource': 'core', 'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'vary': 'Accept-Encoding, Accept, X-Requested-With', 'content-encoding': 'gzip', 'x-github-request-id': 'E0B8:4543:19FA8AC:1AF7ACA:64243199'})
草稿文件名：
web自动化测试（python selenium）

**Comment by Fgaoxing** [comment]
这个是说没找到吧，Not Found，你看看你的配置对不对，如果是根目录的话，就不加/



在 2023-03-29 20:41:26，"AuntYang" ***@***.***> 写道：

dev更新日志：
[WARNING] 2023-03-29T12:39:50.437Z 44e60557-b899-4bf2-96e7-8375069b01ea 获取更新成功: 2.6.3
报错日志：
[ERROR] 2023-03-29T12:39:54.231Z db25e1ac-338f-472d-944c-bee0561afe14 UnknownObjectException(404, {'message': 'Not Found', 'documentation_url': 'https://docs.github.com/rest/reference/repos#get-repository-content'}, {'server': 'GitHub.com', 'date': 'Wed, 29 Mar 2023 12:39:54 GMT', 'content-type': 'application/json; charset=utf-8', 'transfer-encoding': 'chunked', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': '', 'x-github-media-type': 'github.v3; format=json', 'x-github-api-version-selected': '2022-11-28', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '4972', 'x-ratelimit-reset': '1680096378', 'x-ratelimit-used': '28', 'x-ratelimit-resource': 'core', 'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'vary': 'Accept-Encoding, Accept, X-Requested-With', 'content-encoding': 'gzip', 'x-github-request-id': 'E0B8:4543:19FA8AC:1AF7ACA:64243199'})
草稿文件名：
web自动化测试（python selenium）

**Reply by AuntYang** [reply]
但我其他文章能够正常读写

**Comment by Fgaoxing** [comment]
截图博客配置信息（token打马）

**Reply by AuntYang** [reply]
# Hexo Configuration
## Docs: https://hexo.io/docs/configuration.html
## Source: https://github.com/hexojs/hexo/

# Site
title: 梅园小径
subtitle: 
description: 'auntyang的生活与技术小屋'
keywords:
author: auntyang
language: zh-CN
timezone: 'Asia/Shanghai'

# URL
## Set your site url here. For example, if you use GitHub Page, set url as 'https://username.github.io/project'
url: 
permalink: :year/:month:day:abbrlink.html
abbrlink:
  alg: crc16 #算法选项：crc16丨crc32
  rep: dec #输出进制：dec为十进制，hex为十六进制
permalink_defaults:
pretty_urls:
  trailing_index: true # Set to false to remove trailing 'index.html' from permalinks
  trailing_html: true # Set to false to remove trailing '.html' from permalinks

# Directory
source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:

# Writing
new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link:
  enable: true # Open external links in new tab
  field: site # Apply to the whole site
  exclude: ''
filename_case: 0
render_drafts: false
post_asset_folder: true
relative_link: false
future: true
highlight:
  enable: false
  line_number: true
  auto_detect: false
  tab_replace: ''
  wrap: true
  hljs: false
prismjs:
  enable: true
  preprocess: true
  line_number: true
  tab_replace: ''

# Home page setting
# path: Root path for your blogs index page. (default = '')
# per_page: Posts displayed per page. (0 = disable pagination)
# order_by: Posts order. (Order by date descending by default)
index_generator:
  path: ''
  per_page: 12
  order_by: -date

# Category & Tag
default_category: uncategorized
category_map:
tag_map:

# Metadata elements
## https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta
meta_generator: true

# Date / Time format
## Hexo uses Moment.js to parse and display date
## You can customize the date format as defined in
## http://momentjs.com/docs/#/displaying/format/
date_format: YYYY-MM-DD
time_format: HH:mm:ss
## updated_option supports 'mtime', 'date', 'empty'
updated_option: 'mtime'

# Pagination
## Set per_page to 0 to disable pagination
per_page: 12
pagination_dir: page

# Include / Exclude file(s)
## include:/exclude: options only apply to the 'source/' folder
include:
exclude:
ignore:

# Extensions
## Plugins: https://hexo.io/plugins/
## Themes: https://hexo.io/themes/
theme: hexo-theme-matery

# Deployment
## Docs: https://hexo.io/docs/one-command-deployment
deploy:
  type: git
  repo: git@github.com:yepoda/yepoda.github.io.git
  branch: master
  
# 内容搜索插件
search:
  path: search.xml
  field: post
  content: true
  format: html

# rss订阅支持
## 在 public 文件夹中即可看到 atom.xml 文件，说明已经安装成功了。
feed:
  type: atom
  path: atom.xml
  limit: 20
  hub:
  content:
  content_limit: 140
  content_limit_delim: " "
  order_by: -date

# 豆瓣卡片
## 
doubanCard:
  enable: true
  cookie: 
  imgProxy: https://images.weserv.nl/?url=
  
# 豆瓣页面 
# hexo-douban
douban:
  id:  # 豆瓣id
  builtin: true
  book:
    path: books/index.html
    title: '我的书籍清单'
    quote: '思想和身体，总有一个在路上'
  movie:
    path: movies/index.html
    title: '我的电影剧集清单'
    quote: '过去、现在、和未来'
  timeout: 10000

**Reply by AuntYang** [reply]
注释识别成md了
